src/main/java/com/google/step/youtube/DescriptionParser.java

```diff
@@ -127,7 +129,7 @@ private static String getBoundedSnippet(int targetIndex, String description) {
         // add 1 to not include starting space in snippet
         int startIndexByWord = description.indexOf(" ", targetIndex - MAX_SNIPPET_LENGTH / 2) + 1;
         int endIndexByWord = description.lastIndexOf(" ", targetIndex + MAX_SNIPPET_LENGTH / 2);
-        return description.substring(startIndexByWord, endIndexByWord);
+        return "... " + description.substring(startIndexByWord, endIndexByWord) + " ...";
```

Comment:
What if there is not supposed to be ... before the String?

Reply:
This would only happen if the part of the paragraph after the promocode was longer than 100 chars. I just made the fix